File: JPASquadGoals/src/main/java/com/skilldistillery/squadgoals/entities/Review.java
package com.skilldistillery.squadgoals.entities;

import java.time.LocalDateTime;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.MapsId;

@Entity
public class Review {
	
	@EmbeddedId
	private ReviewId id;
	
	private int rating;
	private String comment;
	
	@Column(name="review_date")
	private LocalDateTime reviewDate;
	
	@ManyToOne
	@JoinColumn(name="user_id")
	@MapsId(value = "userId")
	private User user;
	
	@ManyToOne
	@JoinColumn(name="goal_id")
	@MapsId(value = "goalId")
	private Goal goal;

	public ReviewId getId() {
		return id;
	}

	public void setId(ReviewId id) {
		this.id = id;
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public Goal getGoal() {
		return goal;
	}

	public void setGoal(Goal goal) {
		this.goal = goal;
	}

	public Review() {
		
	}
	
	public LocalDateTime getReviewDate() {
		return reviewDate;
	}
	
	public void setReviewDate(LocalDateTime reviewDate) {
		this.reviewDate = reviewDate;
	}

	public int getRating() {
		return rating;
	}

	public void setRating(int rating) {
		this.rating = rating;
	}

	public String getComment() {
		return comment;
	}

	public void setComment(String comment) {
		this.comment = comment;
	}
	
	
}

File: ngSquadGoals/src/app/models/review.ts
export class Review {
  id: any = null; //**VERIFY** EMBEDDED ID
  rating: number = 0;
  comment: string = '';
  reviewDate: Date | null = null; //dates not initialized
  user: any = null;
  goal: any = null;
  images: any[] = [];

  constructor(
    id: any = null, //**VERIFY** EMBEDDED ID
  rating: number = 0,
  comment: string = '',
  reviewDate: Date | null = null,
  user: any = null,
  goal: any = null,
  images: any[] = []
  ) {
  this.id = id;
  this.rating = rating;
  this.comment = comment;
  this.reviewDate = reviewDate;
  this.user = user;
  this.goal = goal;
  this.images = images;
  }
}
